src/services/codecovEventMetrics/useStoreCodecovEventMetric.tsx

```diff
@@ -0,0 +1,103 @@
+import { useMutation } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+
+import Api from 'shared/api'
+
+const mutationQuery = `
+  mutation storeEventMetric(
+    $input: StoreEventMetricsInput!
+  ) {
+    storeEventMetric(
+      input: $input
+    ) {
+      error {
+        ... on UnauthenticatedError {
+          message
+        }
+        ... on ValidationError {
+          message
+        }
+      }
+    }
+  }
+`
+
+interface Params {
+  provider: string
+}
+
+interface StoreEventMetricMutationArgs {
+  owner: string
+  event: string
+  jsonPayload: object
+}
+
+const LOCAL_STORAGE_KEY = 'UserOnboardingMetricsStored'
+const MAX_ENTRIES = 30
+
+const generateMetricString = (
+  owner: string,
+  event: string,
+  jsonPayload: string
+) => {
+  return `${owner}|${event}|${jsonPayload}`
+}
+
+const getStoredMetrics = (): string[] => {
+  const storedMetrics = localStorage.getItem(LOCAL_STORAGE_KEY)
+  return storedMetrics ? JSON.parse(storedMetrics) : []
+}
+
+const isMetricInLocalStorage = (metricString: string) => {
+  const storedMetrics = getStoredMetrics()
+  return storedMetrics.includes(metricString)
+}
+
+const addMetricToLocalStorage = (metricString: string) => {
+  let storedMetrics = getStoredMetrics()
+
+  // limit size of metrics stored to ensure we are in reasonable localstorage limits
+  if (storedMetrics.length >= MAX_ENTRIES) {
+    storedMetrics = storedMetrics.slice(1)
+  }
+
+  storedMetrics.push(metricString)
+  localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(storedMetrics))
+}
```

Comment:
what is the purpose of adding metrics to local storage? Was thinking maybe for batching requests, but it doesn't seem like that's what we're doing here

Reply:
It's so we don't call GQL multiple times unnecessarily. Once something is logged, we store the fact that it _was_ logged and do not make additional network calls.

Reply:
Right now we are only storing the first instance of events(using `get_or_create` on the backend) so it will be a redundant call. Intentionally doing this both to reduce a lot of DB writes as well as focusing on what's important in the user journey. 

This can be extended in the future by customizing payload information...but I think we'd need to revisit the design to see if just using postgres is scalable for our use case if we are doing a large amount of collection.

Reply:
Ok sounds good. Another question I just thought of:

How do we plan to handle data from existing orgs? 

For example, this code gets merged, Codecov org is already onboarded, but the next time we setup a new repo, these data points will be recorded for the first time. Not necessarily something we need to have an answer for right now, more just something that could be important to keep in mind as we build our queries.

Reply:
We could do a join on the org table and filter by creation date. 

It will need some custom querying for sure so I will raise that point when sharing with the team, thanks.